webapp/channels/src/components/announcement_bar/text_dismissable_bar.tsx

```diff
@@ -20,67 +20,62 @@ interface Props extends Partial<AnnouncementBarProps> {
     className?: string;
 }
 
-type State = {
-    dismissed: boolean;
-}
+const options = {
+    singleline: true,
+    mentionHighlight: false,
+};
 
-export default class TextDismissableBar extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
+const getDismissed = (text?: React.ReactNode) => localStorage.getItem(localStoragePrefix + text?.toString()) === 'true';
 
-        this.state = {
-            dismissed: true,
-        };
-    }
+const TextDismissableBar = ({
+    allowDismissal,
+    text,
+    onDismissal,
+    ...extraProps
+}: Props) => {
+    const [dismissed, setDismissed] = useState<boolean>(() => getDismissed(text));
 
-    static getDerivedStateFromProps(props: Props) {
-        const dismissed = localStorage.getItem(localStoragePrefix + props.text?.toString());
-        return {
-            dismissed: (dismissed === 'true'),
-        };
-    }
+    useEffect(() => {
+        setDismissed(getDismissed(text));
+    }, [text]);
 
-    handleDismiss = () => {
-        if (!this.props.allowDismissal) {
+    const handleDismiss = useCallback(() => {
+        if (!allowDismissal) {
             return;
         }
         trackEvent('signup', 'click_dismiss_bar');
 
-        localStorage.setItem(localStoragePrefix + this.props.text?.toString(), 'true');
-        this.setState({
-            dismissed: true,
-        });
-        if (this.props.onDismissal) {
-            this.props.onDismissal();
+        localStorage.setItem(localStoragePrefix + text?.toString(), 'true');
+        setDismissed(
+            true,
+        );
+        if (onDismissal) {
+            onDismissal();
         }
```

Comment:
We can remove the if here if we use the `?` operator.
```suggestion
        onDismissal?.();
```